tamagotchi/src/contract.rs

```diff
@@ -0,0 +1,58 @@
+use gear_lib::non_fungible_token::{io::NFTTransfer, nft_core::*, state::*, token::*};
+use gear_lib_derive::{NFTCore, NFTMetaState, NFTStateKeeper};
+use gmeta::Metadata;
+use gstd::{errors::Result as GstdResult, exec, msg, prelude::*, ActorId, MessageId};
+use hashbrown::HashMap;
+use primitive_types::{H256, U256};
+use program_io::{
+    ProgramMetadata,
+    Tamagotchi,
+    TmgAction,
+    TmgEvent
+};
+
+static mut TAMAGOTCHI: Option<Tamagotchi> = None;
+
+fn state_mut() -> &'static mut Tamagotchi {
+    let state = unsafe { TAMAGOTCHI.as_mut() };
+
+    debug_assert!(state.is_some(), "state isn't initialized");
+
+    unsafe { state.unwrap_unchecked() }
+}
+
+#[no_mangle]
+unsafe extern "C" fn init() {
```

Comment:
Better is to not use `unsafe` in function.
```suggestion
extern "C" fn init() {
```